shell/platform/windows/system_utils.cc

```diff
@@ -25,20 +26,46 @@ std::vector<LanguageInfo> GetPreferredLanguageInfo() {
 
 std::vector<std::wstring> GetPreferredLanguages() {
   std::vector<std::wstring> languages;
+  BOOL languages_from_registry = TRUE;
+  ULONG buffer_size = 0;
+  ULONG count = 0;
   DWORD flags = MUI_LANGUAGE_NAME | MUI_UI_FALLBACK;
 
-  // Get buffer length.
-  ULONG count = 0;
-  ULONG buffer_size = 0;
-  if (!::GetThreadPreferredUILanguages(flags, &count, nullptr, &buffer_size)) {
-    return languages;
+  // Determine where languages are defined and get buffer length
+  if (RegGetValueA(HKEY_CURRENT_USER,
+                   "Control panel\\International\\User Profile", "Languages",
+                   RRF_RT_REG_MULTI_SZ, NULL, NULL, &buffer_size) != S_OK) {
+    languages_from_registry = FALSE;
+    if (!::GetThreadPreferredUILanguages(flags, &count, nullptr,
+                                         &buffer_size)) {
+      return languages;
+    }
   }
 
-  // Get the list of null-separated languages.
+  // Mutli-string must be at least 3-long if non-empty,
+  // as a mulit-string is terminated with 2 nulls.
+  if (buffer_size < 3) {
+    languages_from_registry = FALSE;
+  }
+
+  // Initialize the buffer
   std::wstring buffer(buffer_size, '\0');
-  if (!::GetThreadPreferredUILanguages(flags, &count, buffer.data(),
-                                       &buffer_size)) {
-    return languages;
+  if (languages_from_registry) {
+    std::string str_buffer(buffer_size, '\0');
+    if (RegGetValueA(HKEY_CURRENT_USER,
+                     "Control panel\\International\\User Profile", "Languages",
+                     RRF_RT_REG_MULTI_SZ, NULL, str_buffer.data(),
+                     &buffer_size) != S_OK) {
+      return languages;
+    }
+    for (int i = 0; i < buffer_size; i++) {
```

Comment:
https://github.com/flutter/engine/blob/68f5308d1d9ca7fe021df704b6d5c197ccbe9316/fml/platform/win/wstring_conversion.h#L15-L17

Reply:
`str_buffer` contains a multi-string delimited by nulls, so I believe we would need a `for` loop here no matter what. We could call this function on each substring of the multi-string and increment the counter by its length instead of iterating over each character, but I do not see any benefit to this.